packages/react-dom/src/__tests__/ReactLegacyErrorBoundaries-test.internal.js

```diff
@@ -2124,39 +2142,41 @@ describe('ReactLegacyErrorBoundaries', () => {
   });
 
   // @gate !disableLegacyMode
-  it('discards a bad root if the root component fails', () => {
+  it('discards a bad root if the root component fails', async () => {
     const X = null;
     const Y = undefined;
-    let err1;
-    let err2;
 
-    try {
-      const container = document.createElement('div');
-      expect(() => ReactDOM.render(<X />, container)).toErrorDev(
-        'React.createElement: type is invalid -- expected a string ' +
-          '(for built-in components) or a class/function ' +
-          '(for composite components) but got: null.',
-      );
-    } catch (err) {
-      err1 = err;
-    }
-    try {
-      const container = document.createElement('div');
-      expect(() => ReactDOM.render(<Y />, container)).toErrorDev(
-        'React.createElement: type is invalid -- expected a string ' +
-          '(for built-in components) or a class/function ' +
-          '(for composite components) but got: undefined.',
-      );
-    } catch (err) {
-      err2 = err;
-    }
+    await expect(async () => {
+      await expect(async () => {
+        const container = document.createElement('div');
+        await act(() => {
+          ReactDOM.render(<X />, container);
+        });
+      }).rejects.toThrow('got: null');
+    }).toErrorDev(
+      'Warning: React.jsx: type is invalid -- expected a string ' +
+        '(for built-in components) or a class/function ' +
+        '(for composite components) but got: null.',
+      {withoutStack: 1},
```

Comment:
I don't understand why this switched from `React.createElement` to `React.jsx`?

I assume that's also why there's no component stack either?

Reply:
It wasn't actually asserted before because the render threw which means the assertion wasn't run. The assertion before was asserting the wrong thing.

Reply:
hahahaha awesome